skills_ml/algorithms/occupation_classifiers/base.py

```diff
@@ -63,51 +62,64 @@ def _load_model(self):
         Returns:
             gensim.models.doc2vec.Doc2Vec: The word-embedding model object.
         """
-        if not self.saved:
-            with tempfile.TemporaryDirectory() as td:
-                for f in self.files:
-                    filepath = os.path.join(td, f)
-                    if not os.path.exists(filepath):
+        try:
+            model = Doc2Vec.load('tmp/' + self.model_name)
+            return model
+
+        except:
+            files  = list_files(self.s3_conn, self.s3_path)
+            if not self.saved:
+                with tempfile.TemporaryDirectory() as td:
+                    for f in files:
```

Comment:
While we are modifying this code, could we implement a file lock using, for instance, this module (haven't tried it yet but it looks simple and what we want)? https://pypi.python.org/pypi/filelock

In production, I had problems with the multiprocessing forked processes all trying to read the same file at once. If we used a file lock here, the different processes would all wait for the first process to download the file.